Remove deprecated actions and Playwright API usage

planger opened the pull request:
#### What it does

* Replace usage of deprecated `GabrielBB/xvfb-action@v1`
* Replace usage of deprecated `<obj>.type(...)`

Contributed on behalf of STMicroelectronics.

#### How to test

Ensure workflows and Playwright tests still work as before.

#### Follow-ups

We should update https://github.com/eclipse-theia/theia-playwright-template and https://github.com/eclipse-theia/theia-e2e-test-suite accordingly to avoid using the deprecated GH action.

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

planger commented on .github/workflows/performance-tests.yml:
```diff
@@ -36,14 +36,12 @@ jobs:
-        uses: GabrielBB/xvfb-action@v1
-        with:
-          run: yarn performance:startup:electron
+        shell: bash
```
@xai Do you think omitting xvfb will be a problem for Electron? If yes, we'd need to switch to https://github.com/coactions/setup-xvfb as https://github.com/marketplace/actions/gabrielbb-xvfb-action is deprecated. Thanks!

xai replied:
We need xvfb because we cannot run Electron headless. So yes, we should switch to the maintained action.

planger replied:
Ok, no problem! I updated this PR. Can you please test if this works now? Thank you!

xai replied:
I think we need to test by running the workflow, right?

xai reviewed (approved):
Thank you very much, @planger, for spotting the unmaintained action we are consuming and for fixing our API calls!

Tested fine locally, i.e., the electron tests are still flaky, but no more flaky than before.

I agree that in the Electron case, we need to replace the unmaintained [xvbf-action](https://github.com/GabrielBB/xvfb-action) which had its last real commit in 2021-11.

The [replacement](https://github.com/coactions/setup-xvfb) used in this PR matches the recommendation in the discontinued xvbf-action and it seems to be actively maintained. It also seems to be used frequently (1/10 of the usage that the discontinued action has, which sounds reasonable to me). In the end, by executing the action, we end up executing this file: https://github.com/coactions/setup-xvfb/blob/main/dist/index.js

I very briefly scanned what it does and I have to mention that it includes odd-looking stuff such as [this](https://github.com/coactions/setup-xvfb/blob/6b00cf1889f4e1d5a48635647013c0508128ee1a/dist/index.js#L3454), which is explained with performance optimization. I did not spot any obvious red flags or outright harmful code, but keep in mind that this is a 4K lines javascript file and I looked at it for 3 minutes, which is not much, especially when considering that I am trying to limit my coffee input this week (read: I feel dumb and slow).

I assume that we need to establish a process for vetting third-party actions anyway at some point, and this workflow currently does not even seem to be executed by schedule or dispatch, so I will approve.

marcdumais-work commented:
> The [replacement](https://github.com/coactions/setup-xvfb) used in this PR matches the recommendation in the discontinued xvbf-action and it seems to be actively maintained. It also seems to be used frequently (1/10 of the usage that the discontinued action has, which sounds reasonable to me). In the end, by executing the action, we end up executing this file: https://github.com/coactions/setup-xvfb/blob/main/dist/index.js

Good catch @xai - indeed it's hard to know for sure that this code is not doing anything we would not want it to. I looked at their issue tracker and noticed [this issue](https://github.com/coactions/setup-xvfb/issues/21), that claims that `xvfb` is now part of the Ubuntu 22.04 GitHub runner image, and that in consequence using the action is not longer needed, for that OS. It seems that we only use the action on Ubuntu, so maybe we can forego using that action and still be able to run Electron tests?

marcdumais-work commented on .github/workflows/performance-tests.yml:
```diff
@@ -36,12 +36,12 @@ jobs:
 
       - name: Performance (browser)
-        uses: GabrielBB/xvfb-action@v1
+        uses: coactions/setup-xvfb@v1
```
for the browser, `xvfb` should not be needed? At least when I run this test suite locally, I see no window being created.

xai replied:
I agree, for browser we do not need xvfb.

planger replied:
Right, I think so too. I just kept it in as it was there before. But you're right, let's remove it.

marcdumais-work commented on .github/workflows/performance-tests.yml:
```diff
@@ -36,12 +36,12 @@ jobs:
 
       - name: Performance (Electron)
-        uses: GabrielBB/xvfb-action@v1
+        uses: coactions/setup-xvfb@v1
```
I suggest we give it a try without this action - as I commented on the PR, it would seems `ubuntu-latest` comes with `xvfb` pre-installed.

marcdumais-work replied:
I think these are the same errors today in repo theia-playwright-template, that went away after pinning Python to 3.11:
https://github.com/eclipse-theia/theia/actions/runs/8286508515/job/22676662833

planger replied:
I updated the PR to use `xvfb-run` directly. Thank you for pointing this out!

marcdumais-work commented on .github/workflows/performance-tests.yml:
```diff
@@ -36,12 +36,12 @@ jobs:
-        uses: GabrielBB/xvfb-action@v1
+        uses: coactions/setup-xvfb@v1
         with:
           run: yarn performance:startup:electron
```
ATM this test suite does not seem to work - however it's the same on master, so it's not a problem caused by this PR.

I added trace "analyseTrace(): profilePath ..." - it seems that the trace files, that we attempt to parse using `JSON.parse()`, are binary files not json...

```
$ yarn -s electron rebuild && cd scripts/performance && node electron-performance.js --name 'Electron Frontend Startup' --folder electron --runs 10
native node modules are already rebuilt for electron
analyseTrace(): profilePath: /home/user/theia/scripts/performance/profiles/electron/1.json
[Performance][Electron Frontend Startup][1] Largest Contentful Paint (LCP) failed to obtain a measurement: Unexpected token o in JSON at position 1
Failed to obtain a measurement. The most likely cause is that the performance trace file was incomplete because the script did not wait long enough for "Largest Contentful Paint (LCP)".
SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at analyzeTrace (/home/user/theia/scripts/performance/common-performance.js:151:26)
    at measure (/home/user/theia/scripts/performance/common-performance.js:86:26)
analyseTrace(): profilePath: /home/user/theia/scripts/performance/profiles/electron/2.json
[Performance][Electron Frontend Startup][2] Largest Contentful Paint (LCP) failed to obtain a measurement: Unexpected token o in JSON at position 1
Failed to obtain a measurement. The most likely cause is that the performance trace file was incomplete because the script did not wait long enough for "Largest Contentful Paint (LCP)".
SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at analyzeTrace (/home/user/theia/scripts/performance/common-performance.js:151:26)
    at measure (/home/user/theia/scripts/performance/common-performance.js:86:26)

[...]
```

planger replied:
Thank you very much for looking into this!
I get the same error (in this PR and on master). I think this workflow file has not been used for a long time, also I don't see any runs of this workflow on GH. I just updated it to avoid leaving the deprecated action there.

xai commented:
> I looked at their issue tracker and noticed [this issue](https://github.com/coactions/setup-xvfb/issues/21), that claims that `xvfb` is now part of the Ubuntu 22.04 GitHub runner image, and that in consequence using the action is not longer needed, for that OS. It seems that we only use the action on Ubuntu, so maybe we can forego using that action and still be able to run Electron tests?

Wow, nice! Thanks for checking this! This is really great news! If we have the xvfb-run binary in the image, we can save ourselves a lot of trouble.

planger commented:
Thank you @xai and @marcdumais-work for your thorough review. I've updated the `performance-tests.yml` accordingly. However, as you also say, it doesn't work anyway (also on master) and I just updated it to get rid of the deprecated action and close the security risk, once someone picks up the work on this workflow again.

marcdumais-work commented:
>  I just updated it to get rid of the deprecated action and close the security risk, once someone picks up the work on this workflow again.

I think that's fair. I was curious to confirm whether the image-built-on `xvfb` would work - I think we can trigger a manual run once the PR is merged, to confirm that it's at least present.

marcdumais-work reviewed (approved):
LGTM - thanks @planger !

marcdumais-work commented:
>  I think we can trigger a manual run once the PR is merged, to confirm that it's at least present.

Confirmed, I think: we get the same parsing error I noticed locally, that we briefly discussed : 

https://github.com/eclipse-theia/theia/actions/runs/8296738478/job/22706349277#step:7:20
